LPS25H.class.nut

```diff
@@ -131,120 +174,145 @@ class LPS25H {
-    function setFifoEnable(state) {
-        local val = _readI2C(CTRL_REG2, 1)[0];
-        if (state) {
-            val = val | 0x40;
+        // interrupt pin active-high (active-low by default)
+        local val = _readReg(CTRL_REG3, 1)[0];
+        if (options & INT_ACTIVELOW) {
+            val = val | 0x80;
         } else {
-            val = val & 0xAF;
-        }
-        local res = _writeI2C(CTRL_REG2, val & 0xFF);
-    }
-
-    // -------------------------------------------------------------------------
-    function softReset(state) {
-        local res = _writeI2C(CTRL_REG2, 0x04);
-    }
-
-    // -------------------------------------------------------------------------
-    function setIntActivehigh(state) {
-        local val = _readI2C(CTRL_REG3, 1)[0];
-        if (state) {
             val = val & 0x7F;
-        } else {
-            val = val | 0x80;
         }
-        local res = _writeI2C(CTRL_REG3, val & 0xFF);
-    }
-
-    // -------------------------------------------------------------------------
-    function setIntPushpull(state) {
-        local val = _readI2C(CTRL_REG3, 1)[0];
-        if (state) {
+        // interrupt pin push-pull (open drain by default)
+        if (options & INT_OPENDRAIN) {
+            val = val | 0x40;
+        } else {
             val = val & 0xBF;
+        }
+        // pressure low and pressure high interrupts routed to pin
+        if (enable) {
+            val = val | 0x03;
         } else {
-            val = val | 0x40;
+            val = val & 0xFA;
         }
-        local res = _writeI2C(CTRL_REG3, val & 0xFF);
-    }
+        _writeReg(CTRL_REG3, val & 0xFF);
 
-    // -------------------------------------------------------------------------
-    function setIntConfig(latch, diff_press_low, diff_press_high) {
-        local val = _readI2C(CTRL_REG1, 1)[0];
-        if (latch) {
+        // interrupt latched
+        val = _readReg(INT_CFG, 1)[0] & 0xF8;
+        if (options & INT_LATCH) {
             val = val | 0x04;
         }
-        if (diff_press_low) {
-            val = val & 0x02;
+        // interrupt on low differential pressure
+        if (options & INT_LOW_PRESSURE) {
+            val = val | 0x02;
         }
-        if (diff_press_high) {
+        // interrupt on high differential pressure
+        if (options & INT_HIGH_PRESSURE) {
             val = val | 0x01;
         }
-        local res = _writeI2C(CTRL_REG1, val & 0xFF);
+        _writeReg(INT_CFG, val & 0xFF);
+        
+                
+        // set the enable -----------------------------------------------------
+        val = _readReg(CTRL_REG1, 1)[0];
+        if (enable) {
+            val = val | 0x08;
+        } else {
+            val = val & 0xF7;
+        }
+        _writeReg(CTRL_REG1, val & 0xFF);
+    }
+
+    // -------------------------------------------------------------------------
+    function getInterruptSrc() {
+        return _readReg(INT_SOURCE, 1)[0];
```

Comment:
I'm not a big fan of presenting data this way (I'm also not an embedded developer / ee) - I would rather see us expose the data as a table, it feels more "squirell'y" to me.

Discuss?